docs/middleware/session.md

```diff
@@ -2,142 +2,453 @@
+
+For more details about Fiber v3, see [What’s New](https://github.com/gofiber/fiber/blob/main/docs/whats_new.md).
+
+### Migrating v2 to v3 Example (Legacy Approach)
+
+To convert a v2 example to use the v3 legacy approach, follow these steps:
+
+1. **Initialize with Store**: Use `session.NewStore()` to obtain a store.
+2. **Retrieve Session**: Access the session store using the `store.Get(c)` method.
+3. **Release Session**: Ensure that you call `sess.Release()` after you are done with the session to manage its lifecycle.
 
 :::note
-This middleware uses our [Storage](https://github.com/gofiber/storage) package to support various databases through a single interface. The default configuration for this middleware saves data to memory, see the examples below for other databases.
+When using the legacy approach, the IdleTimeout will be updated when the session is saved.
 :::
 
+#### Example Conversion
+
+**v2 Example:**
+
+```go
+store := session.New()
+
+app.Get("/", func(c *fiber.Ctx) error {
+    sess, err := store.Get(c)
+    if err != nil {
+        return err
+    }
+
+    key, ok := sess.Get("key").(string)
+    if !ok {
+        return c.SendStatus(fiber.StatusInternalServerError)
+    }
+
+    sess.Set("key", "value")
+
+    err = sess.Save()
+    if err != nil {
+        return c.SendStatus(fiber.StatusInternalServerError)
+    }
+
+    return nil
+})
+```
+
+**v3 Legacy Approach:**
+
+```go
+store := session.NewStore()
+
+app.Get("/", func(c *fiber.Ctx) error {
+    sess, err := store.Get(c)
+    if err != nil {
+        return err
+    }
+    defer sess.Release() // Important: Release the session
+
+    key, ok := sess.Get("key").(string)
+    if !ok {
+        return c.SendStatus(fiber.StatusInternalServerError)
+    }
+
+    sess.Set("key", "value")
+
+    err = sess.Save()
+    if err != nil {
+        return c.SendStatus(fiber.StatusInternalServerError)
+    }
+
+    return nil
+})
+```
+
+### v3 Example (Recommended Middleware Handler)
+
+Do not call `sess.Release()` when using the middleware handler. `sess.Save()` is also not required, as the middleware automatically saves the session data.
+
+For the recommended approach, use the middleware handler. See the [Middleware Handler (Recommended)](#middleware-handler-recommended) section for details.
+
+## Types
+
+### Config
+
+Defines the configuration options for the session middleware.
+
+```go
+type Config struct {
+    Storage           fiber.Storage
+    Next              func(c *fiber.Ctx) bool
+    Store             *Store
+    ErrorHandler      func(*fiber.Ctx, error)
+    KeyGenerator      func() string
+    KeyLookup         string
+    CookieDomain      string
+    CookiePath        string
+    CookieSameSite    string
+    IdleTimeout       time.Duration
+    AbsoluteTimeout   time.Duration
```

Comment:
An example in session.md docs would be cool to demonstrate difference between absolute and idle timeouts

Reply:
Like what's new docs